Sources/Sentry/SentryMetricProfiler.mm

```diff
@@ -22,19 +25,19 @@ @interface SentryMetricReading : NSObject
 @implementation SentryMetricReading
 @end
 
-/**
- * Currently set to 10 Hz as we don't anticipate much utility out of a higher resolution when
- * sampling CPU usage and memory footprint, and we want to minimize the overhead of making the
- * necessary system calls to gather that information.
- */
-static const NSTimeInterval kSentryMetricProfilerTimeseriesInterval = 0.1;
-
 NSString *const kSentryMetricProfilerSerializationKeyMemoryFootprint = @"memory_footprint";
 NSString *const kSentryMetricProfilerSerializationKeyCPUUsageFormat = @"cpu_usage_%d";
 
 NSString *const kSentryMetricProfilerSerializationUnitBytes = @"byte";
 NSString *const kSentryMetricProfilerSerializationUnitPercentage = @"percent";
 
+// Currently set to 10 Hz as we don't anticipate much utility out of a higher resolution when
+// sampling CPU usage and memory footprint, and we want to minimize the overhead of making the
+// necessary system calls to gather that information. This is currently roughly 10% of the
+// backtrace profiler's resolution.
+static uint64_t frequencyHz = 10;
+static uint64_t leewayNs = 50000; // 50 milliseconds
```

Comment:
nit: could derive this from `frequencyHz` at runtime instead of hardcoding it (i.e. interval / 2)

Reply:
ah yeah that's what it is in the benchmarker, and i changed it here. i'll put it back to that.